syntax = "proto3";

package auth;

service AuthService {
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
}

message ValidateTokenRequest {
  string token = 1;
}

message ValidateTokenResponse {
  bool valid = 1;
  int32 user_id = 2;
  string email = 3;
  string error = 4;
}

message CreateUserRequest {
  string email = 1;
  string password = 2;
  string full_name = 3;
}

message CreateUserResponse {
  bool success = 1;
  int32 user_id = 2;
  string error = 3;
}

message GetUserRequest {
  int32 user_id = 1;
}

message GetUserResponse {
  int32 id = 1;
  string email = 2;
  string full_name = 3;
  string created_at = 4;
  string error = 5;
}
